elide-model-config/src/main/java/com/yahoo/elide/modelconfig/validator/DynamicConfigValidator.java

```diff
@@ -133,21 +179,57 @@ public static void main(String[] args) {
      */
     public void readAndValidateConfigs() throws IOException {
         this.loadConfigMap();
-        this.setModelVariables(readVariableConfig(Config.MODELVARIABLE));
-        this.setElideSecurityConfig(readSecurityConfig());
-        validateRoleInSecurityConfig(this.elideSecurityConfig);
-        this.setDbVariables(readVariableConfig(Config.DBVARIABLE));
+        this.modelVariables = readVariableConfig(Config.MODELVARIABLE);
+        this.elideSecurityConfig = readSecurityConfig();
+        validateSecurityConfig();
+        this.dbVariables = readVariableConfig(Config.DBVARIABLE);
         this.elideSQLDBConfig.setDbconfigs(readDbConfig());
         this.elideTableConfig.setTables(readTableConfig());
         validateRequiredConfigsProvided();
         validateNameUniqueness(this.elideSQLDBConfig.getDbconfigs());
         validateNameUniqueness(this.elideTableConfig.getTables());
         validateInheritance(this.elideTableConfig);
         populateInheritance(this.elideTableConfig);
-        validateTableConfig(this.elideTableConfig);
+        validateTableConfig();
         validateJoinedTablesDBConnectionName(this.elideTableConfig);
     }
 
+    public void hydrateAndCompileModelConfigs() throws Exception {
```

Comment:
What calls this method?  Why does elide compiler have a singleton method here?

Reply:
this is called from static main method required for cli. Removed this method and creating compiler object in main method.

Reply:
also removed singleton method for compiler class.